validator_client/src/signing_handler.rs

```diff
@@ -111,7 +140,62 @@ impl SigningContext {
     }
 }
 
-impl SigningMethod {
+impl SigningHandler {
+    fn get_from_cache<T: EthSpec, Payload: ExecPayload<T>>(
+        &self,
+        signing_root: &Hash256,
+        message_type: &SignableMessage<'_, T, Payload>,
+    ) -> Option<Signature> {
+        match message_type {
+            SignableMessage::SelectionProof(_) => self
+                .selection_proof_signature_cache
+                .read()
+                .get(signing_root)
+                .map(|val| val.0.clone()),
+            SignableMessage::SyncSelectionProof(_) => self
+                .sync_selection_proof_signature_cache
+                .read()
+                .get(signing_root)
+                .map(|val| val.0.clone()),
+            _ => None,
+        }
+    }
+
+    fn store_in_cache<T: EthSpec, Payload: ExecPayload<T>>(
+        &self,
+        signing_root: Hash256,
+        signature: Signature,
+        message: &SignableMessage<'_, T, Payload>,
+    ) {
+        match message {
+            SignableMessage::SelectionProof(slot) => {
+                let mut cache = self.selection_proof_signature_cache.write();
+                if cache.len() >= MAX_SIGNATURE_CACHE_SIZE {
+                    // Find the entry with the oldest slot and prune it.
+                    let min_slot = cache.iter().min_by_key(|c| c.1 .1);
```

Comment:
I'm not sure if I'm losing my mind, but what is this: `|c| c.1 .1`

Reply:
My current theory is that it's not valid Rust and that Github is giving us a false CI status.

Reply:
bizarre, seems to be a consequence of allowing method chaining

we wouldn't bat an eyelid at

```rust
self.foo()
     .bar
```

nor maybe

```
c.1
  .1
``

but `c    .1    .1` looks weird

Reply:
Oh yeah, make sense!

Reply:
related: https://github.com/rust-lang/rustfmt/issues/4779

Reply:
we could write `|(_, (_, slot))| slot` instead

Reply:
Yep, not sure why `rustfmt` adds the space like that.

> we could write `|(_, (_, slot))| slot` instead

I think this is a good idea :+1: